pyanaconda/ui/tui/spokes/__init__.py

```diff
@@ -248,14 +255,18 @@ class OneShotEditTUIDialog(EditTUIDialog):
     def prompt(self, args=None):
         entry = args
         ret = None
+        self.value = None
 
         if entry:
-            ret = EditTUIDialog.prompt(self, entry)
+            while self.value is None and ret is None:
+                ret = EditTUIDialog.prompt(self, entry)
+
             if ret is None:
                 self.close()
 
         return ret
 
+
 # Inherit abstract methods from NormalTUISpoke
 # pylint: disable=abstract-method
 class EditTUISpoke(NormalTUISpoke):
```

Comment:
Yeah, I know. However, it is little bit more complicated and I think this PR have a really big amount of changes right now.
It would be better to fix what we have now and leave this EditTUISpoke as future change.